workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/ddm-templates/learn-article-template/ddm-template.ftl

```diff
@@ -1,35 +1,64 @@
 <#include "${templatesPath}/SVG">
-
 <script>
 	let href = window.location.href;
-
 	if (href.endsWith("/")) {
 		href = href.substring(0, href.length - 1);
 		window.location.assign(href);
 	}
 </script>
 
+<#assign
+	journalArticleId = .vars["reserved-article-id"].data
+	taxonomyCategoryBriefs = restClient.get("/headless-delivery/v1.0/sites/${groupId}/structured-contents/by-key/${journalArticleId}?nestedFields=embeddedTaxonomyCategory").taxonomyCategoryBriefs
+	taxonomyVocabularies = []
+	taxonomyCategoriesMap = {}
+/>
+
+<#list taxonomyCategoryBriefs as taxonomyCategoryBrief>
+<#assign taxonomyVocabularyName = taxonomyCategoryBrief.embeddedTaxonomyCategory.parentTaxonomyVocabulary.name />
+	<#if !taxonomyVocabularies?seq_contains(taxonomyVocabularyName)>
+		<#assign taxonomyVocabularies = taxonomyVocabularies + [taxonomyVocabularyName] />
+	</#if>
+	<#if taxonomyCategoriesMap[taxonomyVocabularyName]?has_content>
+		<#assign taxonomyCategoriesMap = taxonomyCategoriesMap +
+			{
+			  taxonomyVocabularyName:
+			  taxonomyCategoriesMap[taxonomyVocabularyName] + [{
+				"categoryId": taxonomyCategoryBrief.taxonomyCategoryId,
+				"categoryName": taxonomyCategoryBrief.taxonomyCategoryName
+				}]
+			}
+		/>
+	<#else>
+		<#assign taxonomyCategoriesMap = taxonomyCategoriesMap +
+			{
+			  taxonomyVocabularyName:
+				[{
+					"categoryId": taxonomyCategoryBrief.taxonomyCategoryId,
+					"categoryName": taxonomyCategoryBrief.taxonomyCategoryName
+				}]
+			}
+		/>
+	</#if>
+</#list>
+
 <#assign
 	groupFriendlyURL = "/web" + themeDisplay.getScopeGroup().getFriendlyURL()
 	isLandingPage = false
 	topLevelArticle = true
 />
-
 <#if (breadcrumbLinks.getData())??>
 	<#assign breadcrumbLinksJSONArray = jsonFactoryUtil.createJSONArray(breadcrumbLinks.getData()) />
-
 	<#if breadcrumbLinksJSONArray.length() gt 0>
 		<#assign
 			parentLink = breadcrumbLinksJSONArray.getJSONObject(0)?eval
 			topLevelArticle = false
 		/>
 	</#if>
 </#if>
-
 <#if (landingPage.getData())?? && (landingPage.getData() == "true")>
 	<#assign isLandingPage = true />
 </#if>
-
```

Comment:
are there any reasons to remove this blank line?

Reply:
No, corrected now